src/wp-admin/async-upload.php

```diff
@@ -39,7 +39,8 @@
 }
 
 // just fetch the detail form for that attachment
-if ( isset( $_REQUEST['attachment_id'] ) && ( $id = intval( $_REQUEST['attachment_id'] ) ) && $_REQUEST['fetch'] ) {
+if ( isset( $_REQUEST['attachment_id'] ) && (int) $_REQUEST['attachment_id'] && $_REQUEST['fetch'] ) {
+	$id   = (int) $_REQUEST['attachment_id'];
```

Comment:
Not trying to nit pick. I do not know if this improves anything but do we need to `int()` the `$id` after the check above?

Reply:
Hmm, I see what you mean, that is checked in the `if` statement.

However it is coded that way upstream:
https://github.com/WordPress/wordpress-develop/blob/371966a187081662c155ef98ddb7a681dc10f268/src/wp-admin/async-upload.php#L42

Reply:
I did a little paying and if a `truthy` value is passed as the `attachment_id` - so even the boolean `true` the if evaluates as true to the casting within the if statement is needed as things stand. Really, I think the if check should be an `is_int()` call rather than casting.

Reply:
Seems like this a pattern of work. I have seen this in https://github.com/ClassicPress/ClassicPress/pull/909/files?diff=split&w=0#diff-465c227acbde74516e32664e648c0ac99bce893ec7a16f1a4b2d331ffa7384d1R293-R294 and lines that follow.

Hitting resolved for now.